main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -76,11 +68,21 @@ sealed trait Type {
     */
   def typeConstructors: List[TypeConstructor] = this match {
     case Type.Var(_, _, _, _) => Nil
+    case Type.UnkindedVar(_, _, _) => Nil
     case Type.Cst(tc, _) => tc :: Nil
     case Type.Apply(t1, t2) => t1.typeConstructors ::: t2.typeConstructors
+    case Type.Ascribe(tpe, _) => tpe.typeConstructors
     case Type.Lambda(_, _) => throw InternalCompilerException(s"Unexpected type constructor: Lambda.")
   }
 
+  /**
+    * Returns the base type of a type application.
+    */
+  def baseType: Type.BaseType = this match {
```

Comment:
Move this closer to `.typeConstructor` and describe how it is different.

Sidenote: Where is this used?

Reply:
This is used in the Kinder, where the base type might be a variable and thus have no type constructor.